rllab/dynamics_randomization/mujoco_model_gen.py

```diff
@@ -0,0 +1,127 @@
+import atexit
+from queue import Queue
+from threading import Event
+from threading import Thread
+
+from lxml import etree
+from mujoco_py import load_model_from_xml
+import numpy as np
+
+from rllab.dynamics_randomization import VariationDistribution
+from rllab.dynamics_randomization import VariationMethod
+
+
+class MujocoModelGenerator:
+    """
+    A worker thread to produce to MuJoCo models with randomized dynamic
+    parameters, which are specified by the users of rllab with the class
+    Variations.
+    """
+
+    def __init__(self, file_path, variations):
+        """
+        Starts all the member fields of the class and the worker thread.
+        Parameters
+        ----------
+        file_path : string
+            The absolute path to the XML file that contains the MuJoCo
+            model.
+        variations: Variations
+            An list of Variation objects that indicate the dynamic parameters
+            to randomize in the XML file.
+        """
+        self._variations = variations
+        self._file_path = file_path
+        # Synchronized queue to store mujoco_models
+        self._models = Queue(maxsize=10)
+        # Worker Thread
+        self._worker_thread = Thread(
+            target=self._generator_routine, daemon=True, name="Worker-Thread")
+        # Reference to the generated model
+        self._mujoco_model = None
+        # Communicates the calling thread with the worker thread by awaking
+        # the worker thread so as to generate a new model.
+        self._model_requested = Event()
```

Comment:
This is a great implementation, but I think you can make it even simpler by using the `maxsize` parameter of [`Queue`](https://docs.python.org/3/library/queue.html#queue.Queue)

When maxsize != 0, `put()` blocks when the queue is full, until `get()` is called. By default, `get()` blocks when the queue is empty, until something is `put()` onto the queue.